apps/dcos_dns/src/dcos_dns_config_loader_server.erl

```diff
@@ -71,7 +71,7 @@ handle_info(?REFRESH_MESSAGE, State) ->
         error ->
             {ok, _} = timer:send_after(FailRefreshInterval, ?REFRESH_MESSAGE)
     end,
-    {noreply, State};
+    {noreply, State, hibernate};
```

Comment:
curious: why in certain cases we wait for a certain time before hibernating and why in others we immediately hibernate?

Reply:
I'm trying to avoid gc timer when it's possible, e.g. we call some function periodically, but not too frequently:

`dcos_dns_config_loader_server` sends a message to itself with a 2-minute delay, so it can call garbage collector after each call.

`dcos_dns_mesos` receives a lot of messages from `dcos_net_mesos_listener`, but pushes data to lashup only every n seconds, so it calls garbage collector only when pushes data to lashup. Also, it's just the best moment to call garbage collector because its state is significantly smaller after such push.

`dcos_dns_udp_server` receives udp packets and start new processes, it doesn't have periodic, but not too frequent events. In that case we have to add a gc timer.

Reply:
Got it. Thanks